Packs/RiskSense/Scripts/RiskSenseRansomwareExposureHelperScript/RiskSenseRansomwareExposureHelperScript.py

```diff
@@ -0,0 +1,251 @@
+                    'Description': threat.get('Description', ''),
+                    'Cve': threat.get('Cve', ''),
+                    'Source': threat.get('Source', ''),
+                    'Published': threat.get('Published', ''),
+                    'Updated': threat.get('Updated', ''),
+                    'ThreatLastTrendingOn': threat.get('ThreatLastTrendingOn', ''),
+                    'Trending': threat.get('Trending', '')
+                })
+
+        for vulnerability in vulns:
+            last_trending = vulnerability.get('VulnLastTrendingOn', '')
+            all_cves[vulnerability.get('Cve', 'No ID Found')] = {
+                'Cve': vulnerability.get('Cve', ''),
+                'CVSS': vulnerability.get('BaseScore', ''),
+                'VRR': risk_rating,
+                'ThreatCount': len(threat_list.get(vulnerability.get('Cve', ''), [])),
+                'Trending': vulnerability.get('Trending', ''),
+                'VulnLastTrendingOn': last_trending if last_trending else 'Not Found',
+                'Threats': threat_list.get(vulnerability.get('Cve', ''), []),
+                'Description': vulnerability.get('Description', '')
+            }
+
+    cve_list = []
+    for cve in all_cves.values():
+        if isinstance(cve, list):
+            cve_list.extend(cve)
+        else:
+            cve_list.append(cve)
+
+    return cve_list
+
+
+def get_ransomware_cves():
+    """
+    Get a list of CVEs that contain ransomware threat
+
+    :return: Demisto results entry
+    """
+    response = demisto.get(demisto.context(), 'RiskSense.HostFinding')
+
+    if not response:
+        return 'Key [RiskSense.HostFinding] was not found in context data.', {}, {}
+
+    cve_list = get_cve_list_from_host_findings(response)
+
+    ransomware_cves = []
+    for cve in cve_list:
+        for threat in cve.get('Threats', []):
+            if threat.get('Category', '').lower() == 'ransomware':
+                ransomware_cves.append(cve)
+                break
+
+    ec = {}
+    readable_output = ''
+
+    if len(ransomware_cves) > 0:
+        readable_output += '### Total CVEs found: ' + str(len(ransomware_cves)) + '\n'
+        readable_output += tableToMarkdown('List of CVEs that have ransomware threat', ransomware_cves,
+                                           ['Cve', 'CVSS', 'VRR', 'ThreatCount', 'VulnLastTrendingOn', 'Trending'],
+                                           header_transform, removeNull=True)
+        ec['RiskSense.RansomwareCves(val.Cve && val.Cve == obj.Cve)'] = ransomware_cves
+        return readable_output, ec, {}
+    else:
+        return 'No ransomware CVEs found.', ec, {}
+
+
+def get_ransomware_trending_cves():
+    """
+    Get a list of CVEs which are currently ransomware trending
+
+    :return: Demisto results entry
+    """
+    demisto.executeCommand('DeleteContext', {"key": "Date"})
+    response = demisto.get(demisto.context(), 'RiskSense.HostFinding')
```

Comment:
This is not a good practice in demisto to `DeleteContext` from script, nor call `demisto.context()`, because it means the script depends on some previous command.
instead you either add new input argument and ask the user to provide a value to it.
Or call the command from the script, get the output of it and process.

But in this case, I don't see a reason why not move this script to be a command in the integration, or 3 separate commands

Reply:
sure @anara123 we will definitely fix this and add the same in our best practice to avoid such issue in future.